internal/orgcollector/orgcollector.go

```diff
@@ -0,0 +1,149 @@
+// NOTE: whenever the CC API V3 exposes the quota information for us,
+// we should just store everything in the app cache and get it from there
+package orgcollector
+
+import (
+	"fmt"
+	"sync"
+	"time"
+
+	"github.com/DataDog/datadog-firehose-nozzle/internal/client/cloudfoundry"
+	"github.com/DataDog/datadog-firehose-nozzle/internal/config"
+	"github.com/DataDog/datadog-firehose-nozzle/internal/metric"
+	"github.com/DataDog/datadog-firehose-nozzle/internal/util"
+
+	"github.com/cloudfoundry-community/go-cfclient"
+	"github.com/cloudfoundry/gosteno"
+)
+
+type OrgCollector struct {
+	cfClient         *cloudfoundry.CFClient
+	log              *gosteno.Logger
+	processedMetrics chan<- []metric.MetricPackage
+	customTags       []string
+	queryInterval    uint32
+	stopper          chan bool
+}
+
+func NewOrgCollector(
+	config *config.Config,
+	processedMetrics chan<- []metric.MetricPackage,
+	log *gosteno.Logger,
+	customTags []string) (*OrgCollector, error) {
+	cfClient, err := cloudfoundry.NewClient(config, log)
+	if err != nil {
+		return nil, err
+	}
+	return &OrgCollector{
+		cfClient:         cfClient,
+		log:              log,
+		processedMetrics: processedMetrics,
+		customTags:       customTags,
+		queryInterval:    config.OrgDataQuerySeconds,
+		stopper:          make(chan bool),
+	}, nil
+}
+
+func (o *OrgCollector) Start() {
+	go o.run()
+}
+
+func (o *OrgCollector) Stop() {
+	o.stopper <- true
+}
+
+func (o *OrgCollector) run() {
+	ticker := time.NewTicker(time.Duration(o.queryInterval) * time.Second)
+	for {
+		select {
+		case <-ticker.C:
+			go o.pushMetrics()
+		case <-o.stopper:
+			return
+		}
+	}
+}
+
+func (o *OrgCollector) pushMetrics() {
+	o.log.Info("Collecting org quotas ...")
+	var wg sync.WaitGroup
+	errors := make(chan error, 10)
+	// Fetch orgs
+	wg.Add(1)
+	var allOrgs []cfclient.Org
+	go func() {
```

Comment:
Do we need this go routine since the whole thing is already a go routine?

Reply:
Oh! 🤔 Is it so that we call both `GetV2Orgs` and `GetV2OrgQuotas` in parallel? Statistically speaking I wonder if it is worse it since I doubt that we would have more than 100 orgs in general. What do you think, should we simplify this code in order to gain clarity?

Reply:
I hear that some users have around 1000 orgs or even more, so I think it's worth parallelizing (there will necessarily be at least the same amount of quotas).